### ISSUE TITLE
Imbalanced classes

### ISSUE THREAD

**Issue by skalinin** [bgshih/crnn#86]
Hi,
My dataset is very imbalanced (for example, the class ‘q’ occurs about 100 times in the dataset, but the class ‘a’ may be more than 10 thousand times). What should I do? How can I use class weights in your code?
I think it may looks like this. We have rnn_logits - this is output from RNN, what if I multiply it by class weights before put it in CTC loss? Then CTC loss would have the greater weights for rare classes, and that would impact to backpropagation. Am I right? Could you please help me?

**Comment by githubharald** [comment]
a (character level) language model can do the job. 
You need a decoding algorithm like CTC beam search [1], which combines the output from the RNN and a language model to find the most probable labelling.

[1] I implemented https://arxiv.org/pdf/1601.06581.pdf which works fine.

**Comment by skalinin** [comment]
Thanks for sharing!
Also as I understood, you can create as much data, as you need to (https://github.com/Belval/TextRecognitionDataGenerator)
And from my experience, the question about balancing data not so important (in case of CRNN) as how much data you have. I've needed to generate  a few million images before came closer to solve my problem.
